tests/framework/extensions/provisioning/verify.go

```diff
@@ -335,3 +342,31 @@ func GetSnapshots(client *rancher.Client, localclusterID string, clusterName str
 	return snapshots, nil
 
 }
+
+// VerifySSHTests validates the ssh tests listed in the config on each node of the cluster
+func VerifySSHTests(t *testing.T, client *rancher.Client, clusterObject *steveV1.SteveAPIObject, sshTests []string, clusterID string) {
+	sshUser := clusterObject.JSONResp["sshUser"]
+
+	steveclient, err := client.Steve.ProxyDownstream(clusterID)
+	require.NoError(t, err)
+	nodesSteveObjList, err := steveclient.SteveType("node").List(nil)
+	require.NoError(t, err)
+
+	for _, tests := range sshTests {
+		for _, rancherNode := range nodesSteveObjList.Data {
+			machineName := rancherNode.Annotations["cluster.x-k8s.io/machine"]
+			sshkey, err := sshkeys.DownloadSSHKeys(client, machineName)
+			require.NoError(t, err)
+			assert.NotEmpty(t, sshkey)
+			clusterNode := &nodes.Node{
+				NodeID:          rancherNode.ID,
+				PublicIPAddress: rancherNode.Annotations["rke2.io/external-ip"],
```

Comment:
I just realized something this function is used for both rke2 and k3s. When you first wrote this was only in the rke2 provisioning code, but after the refactor this changed. You would be better off doing this way, and out of the box we also have k3s support.
```
		node := &corev1.Node{}
		err = steveV1.ConvertToK8sType(rancherNode.JSONResp, node)
		require.NoError(t.T(), err)

		ip := nodes.GetNodeIP(node, corev1.NodeExternalIP)
```

 You need these imports 
```
"github.com/rancher/rancher/tests/framework/extensions/kubeapi/nodes"
corev1 "k8s.io/api/core/v1"
```